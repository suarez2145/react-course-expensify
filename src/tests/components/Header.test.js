import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Header from '../../components/HeaderPage';

test( 'should render Header correctly', () => {
    // define 'wrapper varaible to render 'Header' jsx component
    const wrapper = shallow(<Header />)
    // the toJSON takes all the un-important enzyme internal code that gets rendered in the snapshot file and gets rid of it 
    // leaving only the snapshot our our component  
    expect(toJSON(wrapper)).toMatchSnapshot();

    // below is example of what enzyme can do with our tests and check specific portions of our UI
    // same as last tests we expect... but here we can use enzymes built in functions like '.find' and '.text'
    // to check different aspects of our 'Header' component
    // expect(wrapper.find('h1').text('Expensify App'));


    // below is the default React renderer
    // const renderer = new ReactShallowRenderer();
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
})