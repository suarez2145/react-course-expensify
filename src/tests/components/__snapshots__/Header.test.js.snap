// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify App
  </h1>
  <ul>
    <li>
      <NavLink
        activeClassName="is-active"
        exact={true}
        to="/"
      >
         Dashboard 
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        to="/create"
      >
         Create expense 
      </NavLink>
    </li>
  </ul>
</header>
`;
